wgpu-hal/src/gles/egl.rs

```diff
@@ -1475,9 +1509,17 @@ impl crate::Surface for Surface {
         };
 
         if let Some(window) = wl_window {
-            let library = &self.wsi.display_owner.as_ref().unwrap().library;
+            let library = &self
+                .wsi
+                .display_owner
+                .as_ref()
+                .ok_or(crate::SurfaceError::Other("No WSI display owner"))?
+                .library;
             let wl_egl_window_resize: libloading::Symbol<WlEglWindowResizeFun> =
-                unsafe { library.get(c"wl_egl_window_resize".to_bytes()) }.unwrap();
+                unsafe { library.get(c"wl_egl_window_resize".to_bytes()) }.map_err(|err| {
+                    log::error!("Failed to load `wl_egl_window_resize`: {err}");
+                    crate::SurfaceError::Other("Failed to load `wl_egl_window_resize")
```

Comment:
logging the error we're reporting anyways seems redundant. Hard to tell what's the criteria for log + error return versus just error return
```suggestion
                unsafe { library.get(c"wl_egl_window_resize".to_bytes()) }.map_err(|err| {
                    crate::SurfaceError::Other("Failed to load `wl_egl_window_resize")
```